filestore/filestore.go

```diff
@@ -18,7 +18,7 @@ func NewLocalFileStore(basedirectory string) (FileStore, error) {
 			break
 		}
 	}
-	base := basedirectory[0:i + 1]
+	base := basedirectory[0 : i+1]
```

Comment:
There's like `path.Clean`, `path.Join` that can do some of the path manipulation in this file...